src/main.cpp

```diff
@@ -20,6 +21,21 @@ int main()
         sprite.setPosition((float)window.getSize().x / 2, (float)window.getSize().y / 2);
     }
 
+    sf::Font font;
+    if (!font.loadFromFile("resources/OpenSans-Regular.ttf"))
+    {
+        
+        std::cerr << "Failed to load font. Application will continue without displaying text.\n";
+    }
+    else
+    {
+        sf::Text text("Hello SFML", font, 24);
```

Comment:
This `text` object will be destroyed as soon as the `else` block ends. You probably need to store a pointer to `Text` that remains `nullptr` by default and gets initialized with a valid `Text` object only when the font loads correctly.

For example,
```cpp
sf::Text* text = nullptr;

sf::Font font;
if (font.loadFromFile("resources/OpenSans-Regular.ttf"))
{
    text = new sf::Text("Hello SFML", font, 24);
    text->setFillColor(sf::Color::White);
    ... // Text setup
}
else
{
   ... // Error logging
}

...

while (window.isOpen())
{
    ...
    if (text)
    {
        window.draw(*text);
    }
}

delete text;
```